modules/ballerina-jms/src/test/java/org/ballerinalang/nativeimpl/util/BTestUtils.java

```
@@ -0,0 +1,213 @@
+     * 
+     * @param result Result from compilation
+     * @param errorIndex Index of the error in the result
+     * @param expectedErrMsg Expected error message
+     * @param expectedErrLine Expected line number of the error
+     * @param expectedErrCol Expected column number of the error
+     */
+    public static void validateError(CompileResult result, int errorIndex, String expectedErrMsg, int expectedErrLine,
+                                   int expectedErrCol) {
+        Diagnostic diag = result.getDiagnostics()[errorIndex];
+        Assert.assertEquals(diag.getMessage(), expectedErrMsg, "incorrect error message:");
+        Assert.assertEquals(diag.getPosition().getStartLine(), expectedErrLine, "incorrect line number:");
+        Assert.assertEquals(diag.getPosition().startColumn(), expectedErrCol, "incorrect column position:");
+    }
+
+    private static ProgramFile getExecutableProgram(org.wso2.ballerinalang.programfile.ProgramFile programFile) {
+        ByteArrayInputStream byteIS = null;
+        ByteArrayOutputStream byteOutStream = new ByteArrayOutputStream();
+        try {
+            ProgramFileWriter.writeProgram(programFile, byteOutStream);
+
+            // Populate the global scope
+            BLangPrograms.loadBuiltinTypes();
+
+            // Populate the native function/actions
+            BLangPrograms.populateNativeScope();
+
+            ProgramFileReader reader = new ProgramFileReader();
+            byteIS = new ByteArrayInputStream(byteOutStream.toByteArray());
+            return reader.readProgram(byteIS);
+        } catch (IOException e) {
+            throw new RuntimeException(e.getMessage(), e);
+        } finally {
+            if (byteIS != null) {
+                try {
+                    byteIS.close();
+                } catch (IOException ignore) {
+                }
+            }
+
+            try {
+                byteOutStream.close();
+            } catch (IOException ignore) {
+            }
```

Shall we log the message here, instead of silently ignoring ?